### PR TITLE
APGIOV-26081 - refresh cache

### PR DESCRIPTION
**Pull request by sbolosan**
1. when agent status updates, check to see if 7 days past the last refresh cache and if yes, rebuild. If no, NOP
  a. on startup
  b. on status updates
3. (regression) - keep rebuild cache process on error

### CODE REVIEW

**Comment by sbolosan** [comment]
@jcollins-axway @vivekschauhan will add logging around when we call rebuild cache

File: pkg/agent/agent.go [hunk 1]
```diff
@@ -427,6 +434,11 @@ func GetAgentResource() *apiV1.ResourceInstance {
 	return agent.agentResourceManager.GetAgentResource()
 }
 
+// GetAgentResource - Returns Agent resource
+func GetAgentResourceManager() resource.Manager {
+	return agent.agentResourceManager
+}
```
Comment (jcollins-axway) [inline pkg/agent/agent.go:440]: GetResourceManager() resource.Manager
vs
GetAgentResourceManager() resource.Manager

?

Comment (sbolosan) [reply pkg/agent/agent.go:440]: definitely can do.

File: pkg/agent/eventsync.go [hunk 2]
```diff
@@ -132,11 +135,40 @@ func (es *EventSync) initCache() error {
...
+	} else {
+		// if the cache update time already exists, update to 7 days from current cache update time
+		parsedTime, _ := time.Parse(time.RFC3339, value.(string))
+		parsedTime.Add(7 * 24 * time.Hour)
+		cacheUpdateTime = parsedTime.UnixNano()
+	}
+
+	// persist cacheUpdateTime
+	agentDetails["cacheUpdateTime"] = strconv.FormatInt(cacheUpdateTime, 10)
+	agent.apicClient.CreateSubResource(agentInstance.ResourceMeta, map[string]interface{}{definitions.XAgentDetails: agentDetails})
```
Comment (jcollins-axway) [inline pkg/agent/eventsync.go:170]: Rather then getting the details from the, creating the map above if necessary, then adding to the map, you can use the util method to help.

```
agentInstance := agent.agentResourceManager.GetAgentResource()
util.SetAgentDetailsKey(agentInstance, "cacheUpdateTime", strconv.FormatInt(cacheUpdateTime, 10))
agent.apicClient.CreateSubResource(agentInstance.ResourceMeta,map[string]interface{}{definitions.XAgentDetails: util.GetAgentDetails(agentInstance})
```

Comment (jcollins-axway) [reply pkg/agent/eventsync.go:170]: Maybe make a util method that gets just the details subres, returning this.

`map[string]interface{}{definitions.XAgentDetails: util.GetAgentDetails(agentInstance}`

Comment (sbolosan) [reply pkg/agent/eventsync.go:170]: can do...

**Review by jcollins-axway — DISMISSED** [review]

**Review by vivekschauhan — DISMISSED** [review]

**Review by vivekschauhan — APPROVED** [review]

**Pull request merged**